File: algorithms/python/matrix.py
import sys
import time

def matmul(A, B, n):
    C = [[0] * n for _ in range(n)]
    for i in range(n):
        for k in range(n):
            aik = A[i][k]
            for j in range(n):
                C[i][j] += aik * B[k][j]
    return C

def main():
    data = sys.stdin.read().split()
    idx  = 0
    n    = int(data[idx]); idx += 1

    A = []
    for i in range(n):
        row = [int(data[idx+j]) for j in range(n)]
        A.append(row); idx += n

    B = []
    for i in range(n):
        row = [int(data[idx+j]) for j in range(n)]
        B.append(row); idx += n

    start = time.perf_counter()
    C     = matmul(A, B, n)
    ms    = (time.perf_counter() - start) * 1000

    checksum = sum(C[i][j] for i in range(n) for j in range(n))
    sys.stdout.write(f"{checksum}\n")
    sys.stderr.write(f"{ms:.3f}\n")

if __name__ == "__main__":
    main()

File: algorithms/lua/matrix.lua
local function matmul(A, B, n)
    local C = {}
    for i = 1, n do
        C[i] = {}
        for j = 1, n do C[i][j] = 0 end
    end

    for i = 1, n do
        for k = 1, n do
            local aik = A[i][k]
            for j = 1, n do
                C[i][j] = C[i][j] + aik * B[k][j]
            end
        end
    end
    return C
end

local function main()
    local n   = tonumber(io.read("*l"))
    local A, B = {}, {}

    for i = 1, n do
        A[i] = {}
        local line = io.read("*l")
        local j = 1
        for v in line:gmatch("%S+") do
            A[i][j] = tonumber(v); j = j + 1
        end
    end
    for i = 1, n do
        B[i] = {}
        local line = io.read("*l")
        local j = 1
        for v in line:gmatch("%S+") do
            B[i][j] = tonumber(v); j = j + 1
        end
    end

    local start = os.clock()
    local C     = matmul(A, B, n)
    local ms    = (os.clock() - start) * 1000

    local checksum = 0
    for i = 1, n do
        for j = 1, n do checksum = checksum + C[i][j] end
    end

    print(checksum)
    io.stderr:write(string.format("%.3f\n", ms))
end

main()